lib/itamae/resource/file.rb

```diff
@@ -169,6 +176,34 @@ def send_tempfile
           f.unlink if f
         end
       end
+
+      def backup
+        return if !attributes.backup || attributes.backup <= 0
+
+        savetime = Time.now.strftime('%Y%m%d%H%M%S')
+        backup_filename = "#{attributes.path}.itamae-#{savetime}"
+        backup_path = ::File.join(BACKUP_PATH, backup_filename)
+        backup_directory = ::File.dirname(backup_path)
+
+        run_specinfra(:create_file_as_directory, backup_directory)
+        run_specinfra(:copy_file, attributes.path, backup_path)
```

Comment:
Files backed up should have the same permission. `cp -p` preserves the permission of source file.